docs/content/graphql-api-authorization.md

```diff
@@ -0,0 +1,523 @@
+  input.user == hr[_]
+}
+
+# David is the only member of HR.
+hr = [
+  "david",
+]
+```
+
+Build a new bundle with the new policy included.
+
+```shell
+opa build example.rego example-hr.rego
+```
+
+The updated bundle will automatically be served by the bundle server, but note that it  might take up to the configured `max_delay_seconds` for the new bundle to be downloaded by OPA.
+If you plan to make frequent policy changes you might want to adjust this value in `docker-compose.yml` accordingly.
+
+For the sake of the tutorial we included `manager_of` and `hr` data directly inside the policies.
+In real-world scenarios that information would be imported from external data sources.
+
+### 7. Check that the new policy works.
+Check that `david` can see anyone's salary.
+
+```shell
+gql-query david:password "localhost:5000/" '{"query":"query { employeeByID(id: \"alice\") { salary }}"}'
+gql-query david:password "localhost:5000/" '{"query":"query { employeeByID(id: \"bob\") { salary }}"}'
+gql-query david:password "localhost:5000/" '{"query":"query { employeeByID(id: \"charlie\") { salary }}"}'
+gql-query david:password "localhost:5000/" '{"query":"query { employeeByID(id: \"david\") { salary }}"}'
+```
+
+### 8. (Optional) Use JSON Web Tokens to communicate policy data.
+OPA supports the parsing of JSON Web Tokens via the builtin function `io.jwt.decode`.
+To get a sense of one way the subordinate and HR data might be communicated in the real world, let's try a similar exercise utilizing the JWT utilities of OPA.
+
+**example-jwt.rego**:
+
+```live:jwt_example:module:hidden
+package graphqlapi.authz
+
+import future.keywords.in
+import future.keywords.every
+
+query_ast := graphql.parse(input.query, input.schema)[0] # If validation fails, the rules depending on this will be undefined.
+
+# Helper rules.
+
+# Allow users to see the salaries of their subordinates. (variable case)
+allowed_query(q) {
+    selected_salary(q)
+    varname := variable_arg(q, "id")
+    input.variables[varname] in token.payload.subordinates # Do value lookup from the 'variables' object.
+}
+# Allow users to see the salaries of their subordinates. (constant value case)
+allowed_query(q) {
+    selected_salary(q)
+    username := constant_string_arg(q, "id")
+    username in token.payload.subordinates
+}
+
+# Allow users to get their own salaries. (variable case)
+allowed_query(q) {
+    selected_salary(q)
+    varname := variable_arg(q, "id")
+    token.payload.user == input.variables[varname] # Do value lookup from the 'variables' object.
+}
+
+# Allow users to get their own salaries. (constant value case)
+allowed_query(q) {
+    selected_salary(q)
+    username := constant_string_arg(q, "id")
+    token.payload.user == username
+}
+
+# Allow HR members to get anyone's salary.
+allowed_query(q) {
+    selected_salary(q)
+    token.payload.hr == true
+}
+
+# Helper functions.
+
+# Build up a set with all queries of interest as values.
+employeeByIDQueries[value] {
```

Comment:
We could leverage contains and if in these examples,

```
employeeByIDQueries contains value if { # ...
```

Reply:
I slightly prefer the object builder syntax, but I think that's mostly because the object builder syntax is more familiar to me, and that rule is strictly for building a helper/utility object.